File: python/python/src/ConsolePlayer.py
'''
Created on Apr 21, 2017

@author: luke
'''

from Player import Player

class ConsolePlayer(Player):
    '''
    classdocs
    '''

    def selectMove(self, moves, board):
        for i in range(len(moves)):
            print("{}: {}".format(i,moves[i]))
        choice = input("Enter move choice:")
        try:
            intChoice = int(choice)
            if intChoice < 0 or intChoice >= len(moves):
                print("Invalid choice {}".format(intChoice))
                return self.selectMove(moves,board)
            else:
                return moves[intChoice]
        except Exception as e:
            print("Invalid choice {} - {}".format(choice, e))
            return self.selectMove(moves, board)
        
class ConsoleHandler(object):
    def playerTurn(self,whichTurn, currentPlayer, board, color):
        print("Turn {}: Player {} ({})\n{}".format(whichTurn, currentPlayer, color, board))
        
    def gameFinished(self, whichTurn, winningPlayer):
        print("Player {} wins on turn {}".format(winningPlayer, whichTurn))

        

File: ruby/ConsolePlayer.rb
class ConsolePlayer
	def initialize(name)
		@name = name
	end
	def select_move(moves, board)
		moves.each_with_index do |move,index| 
			puts "%d: %s" % [index, move]
		end
		while (true) do
			puts "Enter choice: "
			begin
				choice = gets
				if choice.to_i >= moves.length then
					puts "Invalid choice - too large"
					redo
				end
				move = moves[choice.to_i]
				return move
			rescue => exception
				puts "Invalid choice " + exception.to_s
			end
		end 
	end
	def to_s
		return @name
	end
end

class ConsoleEventHandler
	def playerTurn(whichTurn, currentPlayer, board, color)
		puts "Turn %d: Player %s (%s)\n%s" % [whichTurn, currentPlayer, color, board]
	end
	
	def gameFinished(whichTurn, winningPlayer)
		puts "Player %s wins on turn %d" % [winningPlayer, whichTurn]
	end
end
